src/pages/version/taskDuration/TaskDurationTable.tsx

```diff
@@ -34,46 +33,44 @@ export const TaskDurationTable: React.FC<Props> = ({
   const { sendEvent } = useVersionAnalytics(versionId);
   const { currentStatuses: statusOptions } = useTaskStatuses({ versionId });
 
-  const [
-    {
-      [PatchTasksQueryParams.TaskName]: taskName = "",
-      [PatchTasksQueryParams.Statuses]: statuses = [],
-      [PatchTasksQueryParams.Variant]: variant = "",
-      [PatchTasksQueryParams.Duration]: duration = "",
-    },
-  ] = useQueryParams();
+  const [queryParams, setQueryParams] = useQueryParams();
 
-  const filters = useMemo(
-    () => [
-      { id: PatchTasksQueryParams.TaskName, value: taskName },
-      {
-        id: PatchTasksQueryParams.Statuses,
-        value: Array.isArray(statuses) ? statuses : [statuses],
-      },
-      { id: PatchTasksQueryParams.Variant, value: variant },
-    ],
-    [taskName, statuses, variant]
+  const { initialFilters, initialSort } = useMemo(
+    () => getInitialParams(queryParams),
+    [] // eslint-disable-line react-hooks/exhaustive-deps
   );
 
-  const sorting = useMemo(
-    () => [
-      ...(duration && [
-        {
-          id: PatchTasksQueryParams.Duration,
-          desc: duration === SortDirection.Desc,
-        },
-      ]),
-    ],
-    [duration]
-  );
+  const [filters, setFilters] = useState<ColumnFiltersState>(initialFilters);
+  const [sorting, setSorting] = useState<SortingState>(initialSort);
 
-  const updateQueryParams = useUpdateURLQueryParams();
-  const updateUrl = useCallback(
-    ({ id, value }) => {
-      updateQueryParams({ [id]: value || undefined, page: "0" });
-    },
-    [updateQueryParams]
-  );
+  const updateFilters = (filterState: ColumnFiltersState) => {
```

Comment:
I think at some point here we aren't properly transforming array values for filters which causes them to not be uriencoded correctly.